web/app.js

```diff
@@ -327,34 +327,38 @@ const PDFViewerApplication = {
     if (params.has("verbosity")) {
       AppOptions.set("verbosity", params.get("verbosity") | 0);
     }
-    if (params.has("textlayer")) {
-      switch (params.get("textlayer")) {
-        case "off":
-          AppOptions.set("textLayerMode", TextLayerMode.DISABLE);
-          break;
-        case "visible":
-        case "shadow":
-        case "hover":
-          viewerContainer.classList.add(`textLayer-${params.get("textlayer")}`);
-          try {
-            await loadPDFBug(this);
-            this._PDFBug.loadCSS();
-          } catch (ex) {
-            console.error(`_parseHashParameters: "${ex.message}".`);
-          }
-          break;
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("!GECKOVIEW")) {
```

Comment:
Why do we need to do this exactly, since this code isn't even accessible in a normal Firefox browser without the `pdfjs.pdfBugEnabled` preference being manually set to `true`?
Basically, just leaving this code as-is really can't hurt as far as I'm concerned.